jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/util/HumanTaskHandlerHelper.java

```diff
@@ -201,6 +203,43 @@ public static Deadlines setDeadlines(Map<String, Object> parameters, List<Organi
 		return notifications;
 	}
 
+	protected static void updateDeadineDuration(String durationStr, Deadline deadLine) {
+		// handles iso repetable and both period and date notation
+		try {
+			if (DateTimeUtils.isRepeatable(durationStr)) {
+				String[] repetableParts = DateTimeUtils.parseISORepeatable(durationStr);
+				String tempRepeatLimit = repetableParts[0];
+				String tempDateTimeStr = repetableParts[2];
+
+				if (!tempRepeatLimit.isEmpty()) {
+					deadLine.setRepeatCount(Integer.parseInt(tempRepeatLimit));
+				} else {
+					// just "R/" is accepted
+					deadLine.setRepeatCount(-1);
+				}
+
+				if (DateTimeUtils.isPeriod((tempDateTimeStr))) {
+					deadLine.setDate(new Date(System.currentTimeMillis() + Duration.parse(tempDateTimeStr).toMillis()));
+				} else {
+					deadLine.setDate(new Date(System.currentTimeMillis() + DateTimeUtils.parseDateAsDuration(tempDateTimeStr)));
+				}
```

Comment:
Here you are omitting startDate which is represented by repetableParts[1], i.e. `R/<startDate>/<Duration>` or `R/<startDate>/<endDate>`. In the first if you expect startDate to be now and nothing else. But you can specify startDate and then period, which will result in the deadline date.

In the second if I think you are right here, since if we have end time as a date - we can directly use it.

Reply:
there is a check for repetable above ..should be

Reply:
Yes, but it still doesn't cover case when you have ```R/<startDate>/<Duration>``` In that case the deadline is at startDate+duration time. This code currently covers only a case when startDate is now().

Reply:
start date has always been "now" i didnt change that. will ask in jira if this is needed or not.

Reply:
Yes, I know, but with ISO8601 format it is possible to define it, we already do this with timers here:
https://github.com/kiegroup/jbpm/blob/853baec3573aecf9888658a00a3d7a6b86d56700/jbpm-flow/src/main/java/org/jbpm/process/core/timer/DateTimeUtils.java#L93